ooniprobe/Utility/UIDevice-DeviceInfo.m

```diff
@@ -0,0 +1,25 @@
+#include <sys/sysctl.h>
+#import "UIDevice-DeviceInfo.h"
+
+@implementation UIDevice (DeviceInfo)
+
+- (NSString *)getSysInfoByName:(char *)typeSpecifier
+{
+    size_t size;
+    sysctlbyname(typeSpecifier, NULL, &size, NULL, 0);
+
+    char *answer = malloc(size);
```

Comment:
I could not find in macOS manual pages whether the terminating `\0` is included. According to OpenBSD manual pages it is, but that does not provide us the guarantee that the same happens for Darwin.

Reply:
Is there any change to do here?

Reply:
I think I should do some further research